### DISCUSSION TITLE
inav Pre-arming checks

### DISCUSSION THREAD

**Discussion by sudoflight** [iNavFlight/inav-configurator#2223] [General]
**all** my pre arms are green except hardware health what is that and what could be causing it for quad copter
![Screenshot 2024-10-29 173941](https://github.com/user-attachments/assets/2349b080-596b-474d-8fc9-2e59e823e0e9)

**Comment by Jetrell** [comment]
Hardware health often related to a sensor issue.
Have you looked through the wiki  https://github.com/iNavFlight/inav/wiki/%22Something%22-is-disabled----Reasons

**Reply by sudoflight** [reply]
i am working with micro air f405 v2 fc and a hglrc m80 pro gps are these not compatible.?

**Reply by sudoflight** [reply]
i can get it to arm in this state but quad is slowly fliing in roll after center throttle when it just starts to take off i already adjusted the weight in pid settings when it originally spazzed out 
![1](https://github.com/user-attachments/assets/bba5e0f3-d621-4ce0-9a58-e556021ce898)
![2](https://github.com/user-attachments/assets/cd389e71-49d2-4ecf-a69c-5ad4a533a296)

**Comment by sensei-hacker** [comment]
The hardware status for each piece hardware is indicated by the icons the top of Configurator.

Also in the output of "status", see the "sensor status" line.

You may also see it on the OSD.

**Reply by sudoflight** [reply]
i am working with micro air f405 v2 fc and a hglrc m80 pro gps are these not compatible.?

**Reply by sensei-hacker** [reply]
> i am working with micro air f405 v2 fc and a hglrc m80 pro gps are these not compatible.?


Most likely it's not wired correctly.
Since both the mag AND the GPS are red, I would suspect that either it's not powered on, or the wires are just totally random - possibly because you were unaware wire colors can be completely random. You have to look at which terminals they connect to on the GPS module rather than the color of the wires.

**Reply by sudoflight** [reply]
i can get it to arm in this state but quad is slowly fliing in roll after center throttle when it just starts to take off i already adjusted the weight in pid settings when it originally spazzed out 
![1](https://github.com/user-attachments/assets/c93e19b1-0bd9-4fac-b5f9-1a8ab74db978)
![2](https://github.com/user-attachments/assets/a91050eb-71e6-4b9d-bf38-5541ec050cb8)

**Reply by sudoflight** [reply]
it is on and getting green lite on gps and i am aware of corresponding color wires

**Reply by sudoflight** [reply]
thanks this is helpful

**Reply by sudoflight** [reply]
would that be causeing it to flip

**Reply by sensei-hacker** [reply]
> i can get it to arm in this state but quad is slowly fliing in roll after center throttle when it just starts to take off i already adjusted the weight in pid settings when it originally spazzed out


What flight mode are you in?

Also, you said "when it just starts to take off" - about high high off the ground?  Within about one prop length of the ground (maybe six inches) it's in ground effect and that can cause it to not be very stable.

**Reply by sudoflight** [reply]
on the ground and ch6 arm
![3](https://github.com/user-attachments/assets/b7b04381-f03e-40b8-98cc-1991b88a5588)

**Reply by sudoflight** [reply]
after i get to center stick its good till i throttle up above center some times it rolls left sometime it rolls right

**Reply by sensei-hacker** [reply]
> after i get to center stick its good till i throttle up above center some times it rolls left sometime it rolls right

So before taking off?
I learned the hard way that they can fly, or they can sit on the ground. Trying to do both at the same time doesn't really work.  It's kinda like when my friend was afraid to get on the boat, so as the boat left the dock she had one foot on the boat and one foot on the dock. She ended up between the boat and dock as the boat pulled away.

If it starts to lift off at about half throttle, the way to take off is to quickly go from zero throttle to about 80% or more. To switch from sitting on the ground to flying - 
without trying to do both.

Also if you're fairly new to this, you may want to start with Angle mode.

**Reply by sudoflight** [reply]
k ill have to try tomorrow its dark here now and it was windy all day today

**Reply by sudoflight** [reply]
should i angel with arm same time ? im new to this but not started building back be4 all the regulations stopped for a few years picking it back up

**Reply by sudoflight** [reply]
i use to use apm2.6 and zyx-m  then

**Comment by sensei-hacker** [comment]
Yeah you could do that. Or you could engage Angle before arming if you want.

**Reply by sudoflight** [reply]
things were working but now in pre arm i have a red Navigation is safe

**Reply by sudoflight** [reply]
what may cause and how do i fix

**Reply by sudoflight** [reply]
never mind i got it but having issues with  when im in outputs i have motors thet run off when i use controller i had fixed it b4 and i for got how to fix it

**Comment by sensei-hacker** [comment]
I'm not sure I understand exactly what this means?

> having issues with when im in outputs i have motors thet run off when i use controller

**Reply by sudoflight** [reply]
on inav , in outputs where u can test the motors with transmitter on and armed  isont evn do anything all the motors go up but ine gose to 97-98  then there one that is only about 5 what would be causimg that
